onnxruntime/core/providers/xnnpack/math/gemm.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/xnnpack/xnnpack_kernel.h"
+#include "core/framework/allocator.h"
+#include "core/providers/cpu/math/gemm_base.h"
+#include "core/providers/xnnpack/detail/utils.h"
+#include "core/common/common.h"
+
+namespace onnxruntime {
+class GraphViewer;
+class Node;
+namespace xnnpack {
+
+class Gemm : protected GemmBase, public XnnpackKernel {
+ public:
+  Gemm(const OpKernelInfo& info);
+
+  Status Compute(OpKernelContext* /*context*/) const override;
+
+  static bool IsGemmOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer& graph);
+
+  Status PrePack(const Tensor& tensor, int input_idx, AllocatorPtr alloc,
+                 /*out*/ bool& is_packed,
+                 /*out*/ PrePackedWeights* prepacked_weights) override;
+
+ private:
+  const Tensor* B_{nullptr};
+
+  int64_t M_=-1;
+  int64_t K_=-1;
+  int64_t N_=-1;
+  int64_t numberOfInputs_=-1;
+
```

Comment:
nit: number_of_inputs_

Reply:
replaced the variable with a bool - indicating 3 vs. 2 inputs